File: src/main/java/org/wycliffeassociates/trConverter/BookParser.java
package org.wycliffeassociates.trConverter;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.InputStream;
import java.util.Scanner;

public class BookParser {

    JSONArray arrayOfBooks = new JSONArray();

    public BookParser()
    {
        try {
            InputStream fis = getClass().getClassLoader().getResourceAsStream("assets/books.json");
            String json = ReadInputStream(fis);
            arrayOfBooks = new JSONArray(json);
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }

    public String GetAnthology(String slug)
    {
        String ant = null;

        try {
            for (int i = 0; i < arrayOfBooks.length(); i++) {
                JSONObject jsonBook = arrayOfBooks.getJSONObject(i);
                String bookSlug = jsonBook.getString("slug");

                if(slug.equals(bookSlug))
                {
                    ant = jsonBook.getString("anth");
                }
            }
        }
        catch(JSONException e)
        {
            System.out.printf(e.getMessage());
        }

        return ant;
    }

    public int GetBookNumber(String slug)
    {
        int bn = -1;

        try {
            for (int i = 0; i < arrayOfBooks.length(); i++) {
                JSONObject jsonBook = arrayOfBooks.getJSONObject(i);
                String bookSlug = jsonBook.getString("slug");
                if (slug.equals(bookSlug)) {
                    bn = jsonBook.getInt("num");
                }
            }
        }
        catch(JSONException e)
        {
            System.out.printf(e.getMessage());
        }

        return bn;
    }

    private String ReadInputStream(InputStream fis)
    {
        try (Scanner scanner = new Scanner(fis)) {
            return scanner.useDelimiter("\\A").next();
        }
    }
}

File: src/main/kotlin/org/wycliffeassociates/trConverter/BookParser.kt
package org.wycliffeassociates.trConverter

import org.json.JSONArray
import org.json.JSONException
import java.io.InputStream
import java.util.*

class BookParser {
    internal var arrayOfBooks = JSONArray()

    init {
        try {
            val fis = javaClass.classLoader.getResourceAsStream("assets/books.json")
            val json = readInputStream(fis)
            arrayOfBooks = JSONArray(json)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    fun getAnthology(slug: String): String {
        var ant = ""

        try {
            for (i in 0 until arrayOfBooks.length()) {
                val jsonBook = arrayOfBooks.getJSONObject(i)
                val bookSlug = jsonBook.getString("slug")

                if (slug == bookSlug) {
                    ant = jsonBook.getString("anth")
                }
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return ant
    }

    fun getBookNumber(slug: String): Int {
        var bn = -1

        try {
            for (i in 0 until arrayOfBooks.length()) {
                val jsonBook = arrayOfBooks.getJSONObject(i)
                val bookSlug = jsonBook.getString("slug")
                if (slug == bookSlug) {
                    bn = jsonBook.getInt("num")
                }
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return bn
    }

    private fun readInputStream(fis: InputStream): String {
        Scanner(fis).use { scanner -> return scanner.useDelimiter("\\A").next() }
    }
}